### ISSUE TITLE
DS: Implement Drawer Component in Design System

### ISSUE THREAD

**Issue by dillchen** [hicommonwealth/commonwealth#6005]
## Description

Integrate a new Drawer component into the Design System, ensuring it adheres to our UX and accessibility standards. This component should be versatile and have consistent behavior on desktop or mobile.

Currently the only option is to expand from right side. But may be worth building the component to expand from left. See Screenshot Below

<img width="352" alt="image" src="https://github.com/hicommonwealth/commonwealth/assets/5334557/4ed7a869-9cc4-4ec4-8199-7d71ff4f9da3">

<img width="456" alt="image" src="https://github.com/hicommonwealth/commonwealth/assets/5334557/3c087fb2-04ee-4f72-806b-bf990c83ed87">

## Project Owner
PM: @dillchen 
Designer: @jessmart1213 

## Engineering Requirements
- Develop the Drawer with a responsive design for desktop and mobile platforms.
- Allow product engineers to nest existing DS Components and add data: as props (I.e Reactions Information for a Table in #5509)
- I believe that @mzparacha / @masvelio should discuss how they want the state to be stored and guidelines for integration for PEng. I.e should the drawer state be stored in sidebar state or isolated for each page?

## Acceptance Criteria
- Complete documentation and examples of the Drawer's behavior and implementation in the Design System.
- Added to /components page
- Updates and matches behaviors described in DevLink
- On Click of Double Arrow Should Dismiss the Drawer
- Navigating to a new page should close the drawer

**[DevLink](https://www.figma.com/file/eIVp33a1oCu0AtcLwSbGjr/%F0%9F%9A%A7-Components-and-Patterns?type=design&node-id=3488%3A13&mode=design&t=M5lOIykSs5H0Art2-1)**
[DevLink for Prototype Behavior](https://www.figma.com/proto/eIVp33a1oCu0AtcLwSbGjr/%F0%9F%9A%A7-Components-and-Patterns?type=design&node-id=3488-5831&t=CxBibfswpqOGyIZF-0&scaling=min-zoom&page-id=3488%3A13&starting-point-node-id=3488%3A5831&show-proto-sidebar=1)
[Notion and Loom](https://www.notion.so/buildcommon/Design-solution-Community-Stake-V2-E2E-5035a6f44e6e4744974492ac64b72f8a?pvs=4#7b20c2c96b9d4f8a86934747486350e7)

## Additional context

- This will be the component used for #5509 

Current designs shown below include a click to expand, the desired behavior would be to expand to a whole page upon toggle. However this behavior is not necessarily in scope. The reason for this (as I understand), is we'd need to implement specific pages that detail the exact behavior, and ensure that they're in line with the breadcrumbs. It would be cool if we could some */detail which would take the route as param and show this in a breadcrumb and display the component

**Desktop Behavior In Action**

<img width="822" alt="image" src="https://github.com/hicommonwealth/commonwealth/assets/5334557/22e1e876-24e6-48fb-a472-a52c4fd5ea6a">

**Mobile Behavior In Action**

<img width="307" alt="image" src="https://github.com/hicommonwealth/commonwealth/assets/5334557/fcc2b2b4-ffed-4071-9231-c7710400c159">

----
## Additional Context
- Breadcrumb behavior should be dealt with after Breadcrumb ticket is merged in

**Comment by masvelio** [comment]
@dillchen created PoC for drawer here https://github.com/hicommonwealth/commonwealth/compare/master...marcin.6005-drawer

https://github.com/hicommonwealth/commonwealth/assets/14819225/f3e82422-3246-4159-ac6e-eb5a020c1815

General thoughts:
1. We use MuiBase as a component library, however, they don't have drawer component (yet), so we need to go with 3rd party.
2. There are not too many libraries but for react drawer but after the research I think that [this library](https://www.npmjs.com/package/react-modern-drawer) will be sufficient for us.
3. Full-screen is pretty easy to do if this is what you guys meant (check the video)
4. Regarding state: my opinion is that we should go similar way as we use modals in the app right now, so basically Drawer component should be injected in the particular page and then we should create separate components for DrawerContent. At least my understanding is that we won't have too much drawers within the app, and drawer content would be related to the current page => this is not gonna be global sidebar, right? If you @dillchen and @jessmart1213 could give me examples of every usecase for the drawer (we currently have) and maybe some usecases from the future, then the decision probably will be easier to make.

**Comment by mzparacha** [comment]
Leaving some snippets of my discussion with Marcin.

----

I like the "inject Drawer component in the pages where needed" approach.
Some questions tho
1. I checked the code for react-modern-drawer and it seems like its some simple logic, so do we really need a 3rd party drawer library?
2. We have "Members" and "Groups" tab for the community members page, and based on the active tab in url we show the active state of the page? -- will the drawer have something similar like on url -> /xyz-page?isDrawerOpen=true we load the page and open the drawer immediately -- if yes maybe we can write a common util for it
3. How will we manage page level keyword/focus when the drawer is open (or if we will manage it at all -- currently we dont do that even with modals)? -- i think react-modern-drawer doesn't do that (not sure if there is any other library) .

cc: @masvelio

**Comment by CowMuon** [comment]
I believe this should follow on after table ticket.

**Comment by dillchen** [comment]
Saw things can only be in one milestone (currently in Mobile Friday), not sure if intentional either way, it's blocking #5509 which is a Community Stake V1 Milestone ticket

**Comment by jnaviask** [comment]
@Miaplacidus: Since @masvelio is starting work on the screens for Stake, definitely make sure you two are synced on the status of this ticket and #5509, so he has the components he needs to get the screens working properly.

**Comment by Miaplacidus** [comment]
Points set to 3, since implementation not too complex.
